core/trino-main/src/main/java/io/trino/execution/CommentTask.java

```diff
@@ -151,7 +152,14 @@ private void commentOnColumn(Comment statement, Session session)
             metadata.setViewColumnComment(session, originalObjectName, viewColumn.getName(), statement.getComment());
         }
         else if (metadata.isMaterializedView(session, originalObjectName)) {
-            throw semanticException(TABLE_NOT_FOUND, statement, "Setting comments on the columns of materialized views is unsupported");
+            String columnName = statement.getName().getSuffix();
+            MaterializedViewDefinition materializedViewDefinition = metadata.getMaterializedView(session, originalObjectName).get();
+            ViewColumn viewColumn = materializedViewDefinition.getColumns().stream()
+                    .filter(column -> column.getName().equals(columnName))
+                    .findAny()
+                    .orElseThrow(() -> semanticException(COLUMN_NOT_FOUND, statement, "Column does not exist: %s", columnName));
+            accessControl.checkCanSetColumnComment(session.toSecurityContext(), originalObjectName);
```

Comment:
Are we testing the access control check somewhere ?

Reply:
TestAccessControl.testCommentColumnMaterializedView added